### PR TITLE
Login signup

### PR DESCRIPTION
**Pull request by Tkamunya1**
Authentication still not through

### CODE REVIEW

File: client/src/components/LoginForm.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    fetch("/user_login", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ username, password }),
+    }).then((r) => {
+      setIsLoading(false);
+      if (r.ok) {
+        r.json().then((user) => onLogin(user));
```
Comment (chibukasia) [inline client/src/components/LoginForm.js:24]: We need to redirect a user to different page based on their roles

File: client/src/components/SignUpForm.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
+import React, { useState } from "react";
+import { Button, Error, Input, FormField, Label } from "./styles";
+
+function SignUpForm({ onLogin }) {
+  const [first_name, setFirst_name] = useState("");
+  const [last_name, setLast_name] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
```
Comment (chibukasia) [inline client/src/components/SignUpForm.js:9]: You missed adding the email. Please consider adding an email field and useSate

File: client/src/components/SignUpForm.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        first_name,
+        last_name,
+        username,
+        password,
+        password_confirmation: passwordConfirmation,
+        phone_number,
```
Comment (chibukasia) [inline client/src/components/SignUpForm.js:32]: email is missing

File: client/src/components/SignUpForm.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        role,
+      }),
+    }).then((r) => {
+      setIsLoading(false);
+      if (r.ok) {
+        r.json().then((user) => onLogin(user));
```
Comment (chibukasia) [inline client/src/components/SignUpForm.js:38]: We need to redirect the user after sign up to the appropriate page based on the role. We'll help you with that

File: client/src/components/SignUpForm.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+        <FormField>
+        <Label htmlFor="first_name">First Name</Label>
+        <Input
+          type="text"
+          id="first_name"
+          autoComplete="off"
```
Comment (chibukasia) [inline client/src/components/SignUpForm.js:52]: include the html name attribute as I had told you earlier

File: client/src/components/SignUpForm.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+          value={first_name}
+          onChange={(e) => setFirst_name(e.target.value)}
+        />
+      </FormField>
+        <FormField>
+        <Label htmlFor="last_name">Last Name</Label>
+        <Input
+          type="text"
+          id="last_name"
```
Comment (chibukasia) [inline client/src/components/SignUpForm.js:61]: html name attribute is missing

File: client/src/components/SignUpForm.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+          autoComplete="current-password"
+        />
+      </FormField>
+      <FormField>
+        <Label htmlFor="phone_number">Phone Number</Label>
+        <Input
+          type="text"
+          id="phone_number"
+          onChange={(e) => setPhone_number(e.target.value)}
+        />
```
Comment (chibukasia) [inline client/src/components/SignUpForm.js:113]: Add email input field as well

**Review by chibukasia — CHANGES_REQUESTED** [review]
Please add the comments as well and check the review comments. Also, let me know if you used the react-bootstrap stylings in the styles folder. I may not be familiar with them and I just can't merge something I don't understand. Finally, let commmit and push small chunks of codes. Work on a component, commit and push and lets review it. These has a lot of lines of code to review

**Review by munenebrian — COMMENTED** [review]
I think other than the few pointers from chebukasia, i think it's job well done.

**Review by munenebrian — COMMENTED** [review]
I think other than the few pointers from chebukasia, i think it's job well done.

**Review by chibukasia — APPROVED** [review]

**Pull request merged**